vm-virtio/src/block.rs

```diff
@@ -374,13 +375,11 @@ impl<T: DiskFile> BlockEpollHandler<T> {
         used_count > 0
     }
 
-    fn signal_used_queue(&self) -> result::Result<(), DeviceError> {
+    fn signal_used_queue(&self, queue_index: usize) -> result::Result<(), DeviceError> {
         self.interrupt_status
             .fetch_or(INTERRUPT_STATUS_USED_RING as usize, Ordering::SeqCst);
-        self.interrupt_evt.write(1).map_err(|e| {
-            error!("Failed to signal used queue: {:?}", e);
-            DeviceError::FailedSignalingUsedQueue(e)
-        })
+        (self.interrupt_evt)(self.queues[queue_index].msix_vector);
```

Comment:
This makes the code PCI/MSI specific. See my comment on `IrqClosure`.

Reply:
Yes I agree, I'll rework this.

Reply:
Done